memcached.c

```diff
@@ -10853,34 +10853,47 @@ static void process_sop_command(conn *c, token_t *tokens, const size_t ntokens)
             conn_set_state(c, conn_swallow);
         }
     }
-    else if ((ntokens==5 || ntokens==6) && (strcmp(subcommand, "get") == 0))
+    else if ((ntokens >= 5 && ntokens <= 7) && (strcmp(subcommand, "get") == 0))
     {
+        uint32_t offset = 0;
+        uint32_t count = 0;
         bool delete = false;
         bool drop_if_empty = false;
-        uint32_t count = 0;
 
-        if (! safe_strtoul(tokens[SOP_KEY_TOKEN+1].value, &count)) {
-            print_invalid_command(c, tokens, ntokens);
-            out_string(c, "CLIENT_ERROR bad command line format");
-            return;
-        }
-        if (ntokens == 6) {
-            if (strcmp(tokens[SOP_KEY_TOKEN+2].value, "delete")==0) {
-                delete = true;
-            } else if (strcmp(tokens[SOP_KEY_TOKEN+2].value, "drop")==0) {
+        int read_ntokens = SOP_KEY_TOKEN + 1;
+        int post_ntokens = 1;
+        int rest_ntokens = ntokens - read_ntokens - post_ntokens;
+
+        if (rest_ntokens > 0) {
+            if (strcmp(tokens[read_ntokens+rest_ntokens-1].value, "delete")==0 ||
+                strcmp(tokens[read_ntokens+rest_ntokens-1].value, "drop")==0) {
```

Comment:
수정하는 김에 코드 스타일도 함께 정리해두면 좋겠습니다.

```suggestion
            if (strcmp(tokens[read_ntokens+rest_ntokens-1].value, "delete") == 0 ||
                strcmp(tokens[read_ntokens+rest_ntokens-1].value, "drop") == 0) {
```